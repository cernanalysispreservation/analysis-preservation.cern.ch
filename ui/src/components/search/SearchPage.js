import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import Box from 'grommet/components/Box';

import SearchFacets from './SearchFacets';
import SearchUtils from './SearchUtils';
import SearchResults from './SearchResults';

import {fetchSearch} from '../../actions/search';
import queryString from 'query-string';

class SearchPage extends React.Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.fetchSearch();
  }

  _changePage(page){
    let currentParams = this.getSearchParams();

    const location = {
      search: `${queryString.stringify(Object.assign(currentParams,{page: page}))}`
    };

    this.props.history.push(location);
    this.props.fetchSearch();
  }

  _changePageSize(size){
    let currentParams = this.getSearchParams();
    const location = {
      search: `${queryString.stringify(Object.assign(currentParams,{size: size}))}`
    };

    this.props.history.push(location);
    this.props.fetchSearch();
  }

  componentWillReceiveProps(nextProps) {
    if ( this.props.location.search !== nextProps.location.search ) {
      this.props.fetchSearch()
    }
  }

  getSearchParams() {
    let params = queryString.parse(this.props.location.search);
    return params;
  }

  render() {
    let utils;
    let total = null;
    let results = null;
    let aggs = null;

    let _results = {};
    let _aggs;

    if (this.props.results) {
      _results = this.props.results.toJS();
      _aggs = _results.aggregations;
    }

    if (_aggs) {
      aggs = (
        <SearchFacets
          aggs={_aggs}
          selectedAggs={this.props.selectedAggs}
          onChange={this._toggleAggs}/>
      );
    }

    if (_results && _results.hits) {
      total = _results.hits.total;
      utils = (
        <SearchUtils
          loading={this.props.loading}
          currentPage={this.props.selectedAggs.page ? parseInt(this.props.selectedAggs.page) : 1}
          size={this.props.selectedAggs.size ? parseInt(this.props.selectedAggs.size) : 10}
          total={total || 0}
          onPageChange={this._changePage}
          onPageSizeChange={this._changePageSize}
        />
      );
      results = (<SearchResults results={_results.hits.hits || []} />);
    }

    return (
      <Box  flex={true}>
        {utils}
        <Box flex={true} direction="row">
          {aggs}
          {results}
        </Box>
      </Box>
    );
  }
}

SearchPage.propTypes = {
  loading: PropTypes.bool.isRequired,
  fetchSearch: PropTypes.func,
  history: PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  selectedAggs: PropTypes.array.isRequired,
  results: PropTypes.array.isRequired,
};

function mapStateToProps(state) {
  return {
    results: state.search.getIn(['results']),
    loading: state.search.getIn(['loading']),
    selectedAggs: state.search.getIn(['selectedAggs'])
  };
}

function mapDispatchToProps(dispatch) {
  return {
    fetchSearch: () => dispatch(fetchSearch()),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchPage);