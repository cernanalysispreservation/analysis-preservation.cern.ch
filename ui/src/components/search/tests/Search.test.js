import React from "react";
import { configure, shallow, render, mount } from "enzyme";
import { Map, List, fromJS } from "immutable";
import Adapter from "enzyme-adapter-react-16";

import SearchPage from "../SearchPage";

import { MemoryRouter as Router, Route } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

configure({ adapter: new Adapter() });

let wrapper;
let store;

store = mockStore({
  search: Map({
    query: "",
    aggs: Map({}),
    selectedAggs: Map({}),
    results: Map({
      hits: Map({}),
      links: Map({}),
      aggregations: Map({})
    }),
    error: Map({}),
    loading: false
  })
});
beforeEach(() => {
  wrapper = mount(
    <Provider store={store}>
      <Router>
        <SearchPage />
      </Router>
    </Provider>
  );
});

afterEach(() => {
  wrapper.unmount();
});
describe("Main Header Component", () => {
  it("Should Match", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
