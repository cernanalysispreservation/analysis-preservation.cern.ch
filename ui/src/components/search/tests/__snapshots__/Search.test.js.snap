// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Header Component Should Match 1`] = `undefined`;
